packages/main/src/plugin/provider-registry.ts

```diff
@@ -812,7 +815,10 @@ export class ProviderRegistry {
 
   getMatchingProviderLifecycleContextByProviderId(
     providerId: string,
-    providerConnectionInfo: ProviderContainerConnectionInfo | ProviderKubernetesConnectionInfo,
+    providerConnectionInfo:
+      | ProviderContainerConnectionInfo
+      | ProviderKubernetesConnectionInfo
+      | ContainerProviderConnection,
```

Comment:
This method called only from extension loader and only with objects that is instance of `ContainerProviderConnection', so no union type is needed as well.

```suggestion
    providerConnectionInfo: ContainerProviderConnection,
```